src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private T[] array;
+    private int size = 0;
+
+    public ArrayList() {
+        array = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            // Створюємо новий масив з більшою ємністю
+            T[] newArray = (T[]) new Object[array.length + (array.length / 2)];
+            // Копіюємо елементи в новий масив
+            for (int i = 0; i < size; i++) {
+                newArray[i] = array[i];
+            }
+            array = newArray;
```

Comment:
Consider extracting the array resizing logic into a separate method to adhere to the Single Responsibility Principle. This will also make the `add` method cleaner and more readable.